### PR TITLE
add toleration for Istio components.

### PR DESCRIPTION
**Pull request by morvencao**

### CODE REVIEW

**Comment by sdake** [comment]
@morvencao we are more or less attempting to freeze feature development on this component.  Instead, we are doing work in  https://github.com/istio-ecosystem/istio-installer.  Can you propose an issue tracker there for this feature?  This seems like something that could be handled by kustomize.

Cheers
-steve

**Comment by morvencao** [comment]
@sdake Yep, thanks

**Comment by noony** [comment]
Do you have an alternative if we don't have this patch before istio-installer not in alpha because to get dedicated node for ingress traffic ?

**Comment by morvencao** [comment]
@noony I'm to merge this patch to new installer. For now, we have `nodeSelector` that you can use to deploy the specific component to desired node, see: https://github.com/istio/istio/blob/release-1.1/install/kubernetes/helm/istio/values.yaml#L326 and https://github.com/istio/istio/blob/release-1.1/install/kubernetes/helm/istio/charts/gateways/values.yaml#L116

**Comment by sterchelen** [comment]
> @noony I'm to merge this patch to new installer. For now, we have `nodeSelector` that you can use to deploy the specific component to desired node, see: https://github.com/istio/istio/blob/release-1.1/install/kubernetes/helm/istio/values.yaml#L326 and https://github.com/istio/istio/blob/release-1.1/install/kubernetes/helm/istio/charts/gateways/values.yaml#L116

It is useful to restrict pods scheduling based on toleration. I have a case where daemonset is scheduled on the istio nodes... And the only idiomatic way of not scheduling this daemonset would be to add a taint on the nodes, thus a toleration on the istio installation.

**Comment by morvencao** [comment]
@sterchelen I have created PR for tolerations in new installer: https://github.com/istio/installer/pull/170

**Comment by sdake** [comment]
@morvencao please rebase this on master, then add the mergify flags to auto-backport to 1.2.

As I announced today on discuss, the existing installer will be used for 1.2.  This is a feature, but with real use cases, so I'm open to merging this work now in master and a backport to 1.2 with the understanding that upgrade/downgrade has been tested prior to submission.

Cheers
-steve

**Comment by sdake** [comment]
@noony to your original question, this installer with tolerations should be available in 1.2 without using the alpha-quality `install.next`.

Cheers
-steve

**Comment by morvencao** [comment]
@sdake Thanks, rebased the code. 

> then add the mergify flags to auto-backport to 1.2

How do I enable auto-backport to 1.2 for this PR?

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -329,6 +329,12 @@ global:
   # the desired values.
   defaultNodeSelector: {}
 
+  # Default node tolerations to be applied to all deployments so that all pods can be
```
Comment (linsun) [inline install/kubernetes/helm/istio/values.yaml]: @morvencao is tolerate only used in conjunction with taint? If so, can we make it clear in this comment?

**Comment by linsun** [comment]
@morvencao @sdake FYI, the auto cherry bot won't work with this repo.  Would need to create the cherry pick manually after this one is merged in master.

File: install/kubernetes/helm/istio/values.yaml (hunk 1 above)
Comment (linsun) [reply install/kubernetes/helm/istio/values.yaml]: lgtm otherwise :)

Comment (morvencao) [reply install/kubernetes/helm/istio/values.yaml]: done

**Comment by Pluies** [comment]
Thanks for contributing this @morvencao, removing the special-casing in our chart makes us way closer to upstream, and much easier to maintain!

**Comment by Pluies** [comment]
Actually... Somehow these changes didn't make it to the Istio 1.2.0 release, so we'll have to carry on with our hacks for now :(
Do you think they will be picked up in the 1.2.1 release? Or later?

**Comment by morvencao** [comment]
Cherry-pick this to 1.2.x with https://github.com/istio/istio/pull/15081

**Comment by fpesce** [comment]
> Actually... Somehow these changes didn't make it to the Istio 1.2.0 release, so we'll have to carry on with our hacks for now :(
> Do you think they will be picked up in the 1.2.1 release? Or later?

Hi @Pluies , release manager for 1.2 here,  #15081 has been merged to the 1.2 branch before next minor revision release: it will be picked up for 1.2.1

**Pull request merged**